Cannot create/start entry on fresh install

When installing tid for the first time on a new machine (i.e. `~/.tid` does not exist), a runtime error is caused when attempting to write to the default timesheet. 

### To reproduce

```
mv ~/.tid ~/.tid.bak
tid start foobar
```

### Command Output
```$ tid start foobar
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x500c47]

goroutine 1 [running]:
github.com/SeerUK/tid/vendor/github.com/boltdb/bolt.(*Bucket).Get(0x0, 0xc4200dd458, 0x10, 0x20, 0x10, 0x20, 0xa)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/vendor/github.com/boltdb/bolt/bucket.go:260 +0x37
github.com/SeerUK/tid/pkg/state/bolt.(*boltBackend).Read.func1(0xc4200b4540, 0x640020, 0xc4200b4540)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/bolt/bolt_backend.go:61 +0x100
github.com/SeerUK/tid/vendor/github.com/boltdb/bolt.(*DB).View(0xc4200c4000, 0xc4200dd548, 0x0, 0x0)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/vendor/github.com/boltdb/bolt/db.go:626 +0x9f
github.com/SeerUK/tid/pkg/state/bolt.(*boltBackend).Read(0xc42000e090, 0xc420017510, 0xd, 0xc420017590, 0x10, 0xc420017590, 0x10, 0x10, 0xc420017590, 0xc4200bc240)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/bolt/bolt_backend.go:64 +0xc0
github.com/SeerUK/tid/pkg/state.(*backendStore).Read(0xc4200ce8a0, 0xc420017590, 0x10, 0x72bda0, 0xc4200117d0, 0xc420017590, 0x10)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/store.go:45 +0x6b
github.com/SeerUK/tid/pkg/state.(*storeTrackingGateway).FindTimesheet(0xc4200ce8c0, 0xc420017560, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40, 0x2d36302d37313032)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/tracking_gateway.go:134 +0x1c9
github.com/SeerUK/tid/pkg/state.(*storeTrackingGateway).FindOrCreateTimesheet(0xc4200ce8c0, 0xc420017560, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/tracking_gateway.go:156 +0x96
github.com/SeerUK/tid/pkg/state.(*storeTrackingGateway).FindOrCreateTodaysTimesheet(0xc4200ce8c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/state/tracking_gateway.go:165 +0xfa
github.com/SeerUK/tid/pkg/util.(*TrackingFacade).Start(0xc4200ce8e0, 0x7fff12bebc77, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/util/tracking_facade.go:49 +0xfe
github.com/SeerUK/tid/pkg/tid/cli/command.StartCommand.func2(0xc4200116b0, 0xc4200c3d40, 0xc420013080, 0x55)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/pkg/tid/cli/command/start.go:24 +0x77
github.com/SeerUK/tid/vendor/github.com/eidolon/console.(*Application).Run(0xc420020780, 0xc4200103d0, 0x2, 0x2, 0xc420013080, 0x55, 0x55, 0x0)
	/home/cj/Projects/go/src/github.com/SeerUK/tid/vendor/github.com/eidolon/console/application.go:97 +0x5b3
main.main()
	/home/cj/Projects/go/src/github.com/SeerUK/tid/cmd/tid/main.go:36 +0x290
```

--- a/pkg/state/backend.go
+++ b/pkg/state/backend.go
@@ -22,7 +22,7 @@ var ErrNilBucket = errors.New("state: No bucket found")
 // deal with ProtoBuf messages.
 type Backend interface {
 	// -- Buckets
-	// CreateBucketIfNotExists attempts to create a bucket with a given name, if it doens't exist.
+	// CreateBucketIfNotExists attempts to create a bucket with a given name, if it doesn't exist.
 	CreateBucketIfNotExists(name string) error
 	// HasBucket returns true if this Backend has a bucket with the given name.
 	HasBucket(name string) bool

--- a/pkg/state/bolt/bolt_backend.go
+++ b/pkg/state/bolt/bolt_backend.go
@@ -58,6 +58,11 @@ func (b *boltBackend) Read(bucket string, key string) ([]byte, error) {
 
 	err := b.db.View(func(tx *boltdb.Tx) error {
 		bucket := tx.Bucket([]byte(bucket))
+
+		if bucket == nil {
+			return state.ErrNilBucket
+		}
+
 		value = bucket.Get([]byte(key))
 
 		return nil

--- a/pkg/state/migrate/versions/1496518709.go
+++ b/pkg/state/migrate/versions/1496518709.go
@@ -1,0 +1,45 @@
+package versions
+
+import (
+	"github.com/SeerUK/tid/pkg/state"
+	"github.com/SeerUK/tid/pkg/state/migrate"
+	"github.com/SeerUK/tid/pkg/types"
+	"github.com/SeerUK/tid/pkg/util"
+)
+
+func init() {
+	migrate.RegisterMigration(&Migration1496518709{})
+}
+
+// Migration1496518709 is a backend migration created at 1496518709 unix time.
+type Migration1496518709 struct{}
+
+// Description provides a description of what the migration is doing.
+func (m *Migration1496518709) Description() string {
+	return "Set up status to point to the default workspace if no workspace is set."
+}
+
+// Migrate performs the migration.
+func (m *Migration1496518709) Migrate(backend state.Backend) error {
+	factory := util.NewStandardFactory(backend)
+	sysGateway := factory.BuildSysGateway()
+
+	status, err := sysGateway.FindOrCreateStatus()
+
+	if err != nil {
+		return err
+	}
+
+	if status.Workspace != "" {
+		return nil
+	}
+
+	status.Workspace = types.TrackingStatusDefaultWorkspace
+
+	return sysGateway.PersistStatus(status)
+}
+
+// Version returns the version number of the migration.
+func (m *Migration1496518709) Version() uint {
+	return 1496518709
+}

